be/src/exprs/agg/aggregate_factory.cpp

```diff
@@ -524,6 +542,21 @@ AggregateFuncResolver::AggregateFuncResolver() {
     add_aggregate_mapping<TYPE_DECIMAL64, TYPE_DECIMAL128>("avg");
     add_aggregate_mapping<TYPE_DECIMAL128, TYPE_DECIMAL128>("avg");
 
+    add_aggregate_mapping<TYPE_BOOLEAN, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_TINYINT, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_SMALLINT, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_INT, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_BIGINT, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_LARGEINT, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_FLOAT, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_DOUBLE, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_VARCHAR, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_CHAR, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_DECIMAL32, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_DATETIME, TYPE_ARRAY>("array_agg");
+    add_aggregate_mapping<TYPE_DATE, TYPE_ARRAY>("array_agg");
+    // TYPE_DECIMAL32, TYPE_DECIMAL64, TYPE_DECIMAL128 is not supported now for array_agg
```

Comment:
If we don't support decimal v3, we should add check in FE

Reply:
Fixed